os/base.h

```diff
@@ -0,0 +1,10 @@
+// Stub out Apple internal header file to redirect to linux_base.h.
```

Comment:
thinking about it more, something like this may make the most sense (maybe in a new os/target.h header ?)

    #ifdef __APPLE__
    #include <TargetConditionals.h>
    #else
    #define TARGET_OS_UNIX 1
    #endif

and then conditionalize with `#if TARGET_OS_MAC` resp. `#if TARGET_OS_UNIX` ?